core/api/test/integration/services/lnd-service.spec.ts

```diff
@@ -209,160 +214,281 @@ afterAll(async () => {
+        btcPaymentAmount,
+        maxFeeAmount: undefined,
+      })
+      if (paid instanceof Error) throw paid
+      expect(paid.revealedPreImage).toHaveLength(64)
+      expect(paid.roundedUpFee).toEqual(
+        Number(btcPaymentAmount.amount) * ROUTE_PPM_PERCENT,
+      )
+    })
 
-    const paid = await lndService.payInvoiceViaPaymentDetails({
-      decodedInvoice: lnInvoice,
-      btcPaymentAmount,
-      maxFeeAmount: undefined,
+    it("fails to pay high fee route with max limit set", async () => {
+      const lnInvoice = await createPrivateInvoice({
+        lnd: lndOutside2,
+      })
+
+      const paid = await lndService.payInvoiceViaPaymentDetails({
+        decodedInvoice: lnInvoice,
+        btcPaymentAmount,
+        maxFeeAmount: LnFees().maxProtocolAndBankFee(btcPaymentAmount),
+      })
+      expect(paid).toBeInstanceOf(RouteNotFoundError)
     })
-    expect(paid).toBeInstanceOf(PaymentRejectedByDestinationError)
-  })
 
-  it("pay invoice with High CLTV Delta", async () => {
-    // Create invoice
-    const { request } = await createInvoice({
-      lnd: lndOutside1,
-      tokens: amountInvoice,
-      cltv_delta: 200,
+    it("probes across route with fee higher than payment amount", async () => {
+      const amountInvoice = 1
+
+      // Change channel policy
+      await setFeesOnChannel({
+        localLnd: lndOutside1,
+        partnerLnd: lndOutside2,
+        base: 10,
+        rate: ROUTE_PPM_RATE,
+      })
+
+      // Create invoice
+      const lnInvoice = await createPrivateInvoice({
+        lnd: lndOutside2,
+        tokens: amountInvoice,
+      })
+
+      // Execute probe and check result
+      const probed = await lndService.findRouteForInvoice({
+        invoice: lnInvoice,
+      })
+      expect(probed).toBeInstanceOf(RouteNotFoundError)
+
+      // Reset lnd mission control
+      await deleteForwardingReputations({ lnd: lnd1 })
     })
-    const lnInvoice = decodeInvoice(request)
-    if (lnInvoice instanceof Error) throw lnInvoice
 
-    const paid = await lndService.payInvoiceViaPaymentDetails({
-      decodedInvoice: lnInvoice,
-      btcPaymentAmount,
-      maxFeeAmount: undefined,
+    it("deletes payment", async () => {
+      const { request, secret } = await createInvoice({ lnd: lndOutside1 })
+      const revealedPreImage = secret as RevealedPreImage
+      const lnInvoice = decodeInvoice(request)
+      if (lnInvoice instanceof Error) throw lnInvoice
+      const { paymentHash } = lnInvoice
+
+      const paid = await lndService.payInvoiceViaPaymentDetails({
+        decodedInvoice: lnInvoice,
+        btcPaymentAmount,
+        maxFeeAmount: undefined,
+      })
+      if (paid instanceof Error) throw paid
+
+      // Confirm payment exists in lnd
+      const retrievedPayment = await lndService.lookupPayment({ paymentHash })
+      expect(retrievedPayment).not.toBeInstanceOf(Error)
+      if (retrievedPayment instanceof Error) return retrievedPayment
```

Comment:
this must be a throw

Reply:
https://github.com/GaloyMoney/galoy/pull/3694/commits/dbe2248c4049bc54d967e6c1735eb815ba905dcb